cluster/node.go

```diff
@@ -20,6 +20,59 @@ import (
 	log "github.com/sirupsen/logrus"
 )
 
+type InvalidNodeModeErr string
+
+func (e InvalidNodeModeErr) Error() string {
+	return fmt.Sprintf("invalid cluster operating mode %q", string(e))
+}
+
+//go:generate stringer -type=NodeMode -trimprefix=Mode
+type NodeMode uint8
+
+const (
+	ModeShard NodeMode = iota
+	ModeFull
```

Comment:
This seems like the wrong name for a single node.  "ModeAll" seems like a better fit as the node is acting as both a shard (it ingested data) and a querier.

Reply:
"all" may imply it does "all the things". But in this mode there is no clustering (gossip) participation.
Can you articulate the problem with 'full' ? I suspect you mean because a cluster can have multiple ModeFull nodes, each of which is not fully responsible for the operation of the cluster (though it does get a full view of all the data)

No name is perfect I guess

Reply:
to me, modeFull implies a "full" production deployment.  How it is used here is really a dev single instance.

Given that you shouldnt run with "modeFull" for production, perhaps `modeDev` would be more appropriate.

Reply:
It's a good suggestion, but I think running multiple instances that each consume all data (like we used to run worldping way back) - so without sharding and gossip - is a legitimate way to set up a production cluster.

Note that "mode" has changed meaning from mode of the cluster to the mode of the individual instance, so 'full' means 'the node is full' not 'the cluster is full'

If we can't find a better name by end of today let's keep it.

Reply:
>   but I think running multiple instances that each consume all data (like we used to run worldping way back) - so without sharding and gossip - is a legitimate way to set up a production cluster.

I completely disagree with this.  We only run this topology because MT didnt have gossip back then.  If you have multiple MT instances (which you need for production workloads to allow upgrades and protect against node failures) you definitely should be using gossip.

If users want to do this they still can, but by calling it modeDev we make it clear that a production deployment should be using gossip.

Reply:
> multiple MT instances (which you need for production workloads to allow upgrades and protect against node failures) you definitely should be using gossip.

redundancy for upgrades and protecting against node failures only requires running multiple replicas.
gossip is for sharding which is for load distribution, which is an orthogonal concern.

Reply:
production deployments should use gossip.

just because you can deploy in a different way doesnt mean you should.

With the speculativeQuery feature, even a single shard deployment will benefit from using gossip.

Reply:
That makes sense. we have a legitimate use case for gossip beyond load distribution now, speculative queries across replicas.
I think ModeDev is a good name then, and will change.